src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -316,8 +324,9 @@ int halide_hexagon_remote_run(handle_t module_ptr, handle_t function,
 }
 
 int halide_hexagon_remote_poll_log(char *out, int size, int *read_size) {
+    // Read one line at a time.
     // Leave room for appending a null terminator.
-    *read_size = global_log.read(out, size - 1);
+    *read_size = global_log.read(out, size - 1, '\n');
     out[*read_size - 1] = 0;
```

Comment:
In this case, it's up to the host side. I've been thinking about adding checks to all of the RPC call parameters to prevent crashes on Hexagon. Though, for this specific one, if it fails, then we can't provide any useful information anyways :) The only way to get a message off the DSP is through this function. If the read_size is null, then we probably can't get a message out.